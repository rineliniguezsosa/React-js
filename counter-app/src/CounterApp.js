import PropTypes from 'prop-types'
import React, { useState } from 'react';


export function CounterApp({value}) {

  const [myvalue, setValue] = useState(value)

  const HandleAdd = () =>{
    
  }
  return (
    <>
        <h1>CounterApp</h1>
        <h2>{value}</h2>   

        <button onClick={HandleAdd}>+1</button> 
    </>
  )
}

CounterApp.propTypes = {
    value:PropTypes.number,
}