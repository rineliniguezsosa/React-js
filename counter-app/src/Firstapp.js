import PropTypes from 'prop-types'

export function Firstapp({name,edad}) {
//   const obj = {
//     name:'rinel',
//     email:'rineliniguezsosa@gmail.com'
//   }
  return (
    <>
        {/* <h1>Rinel</h1> */}
        <span>{name}</span>
        <span>{edad}</span>
        {/* <p>{JSON.stringify(obj)}</p> */}
    </>
  )
}

Firstapp.propTypes = {
    name:PropTypes.string,
    edad:PropTypes.number
}

Firstapp.defaultProps = {
    name:'rinel i√±iguez sosa',
    edad:23,
}
