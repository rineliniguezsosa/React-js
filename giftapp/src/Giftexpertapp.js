import React, { useState } from 'react';
import { Addcategoria } from './components/Addcategoria';

export function Giftexpertapp() {
  const [categorias, setCategorias] = useState(['Gon'])

  const addcategory = (value) =>{
    setCategorias([value,...categorias])
  }
  return (
    <>
        <h1>Giftexpertapp</h1>
        <Addcategoria onNewCategorias={(value)=> addcategory(value)}/>
        <button onClick={addcategory}>Agregar una categoria</button>
        {/* LISTAMOS LOS GIFS */}
        <ol>
            {categorias.map(usercategory => (
                <li key={usercategory}>{usercategory}</li>
            ))}
        </ol>
    </>
  )
}
