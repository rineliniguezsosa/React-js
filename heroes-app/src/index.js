import React from 'react';
import ReactDOM from 'react-dom/client';
import { HeroesApp } from './HeroesApp';
import { BrowserRouter } from 'react-router-dom'


const root = ReactDOM.createRoot(document.getElementById('root'));
root.render(
  <React.StrictMode>
   <HeroesApp></HeroesApp>
  </React.StrictMode>
);

