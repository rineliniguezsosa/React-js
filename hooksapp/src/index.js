import React from 'react';
import ReactDOM from 'react-dom/client';
import { SimpleForm } from './02-useEffect/SimpleForm';
//import { Counterapp } from './01-useState/Counterapp';
//import { CounterCustomHook } from './components/CounterCustomHook';
//import { Hookapp } from './components/Hookapp';
import { FormCustomHook } from './components/FormCustomHook'
import './estilos.css'

ReactDOM.createRoot(document.getElementById('root')).render(
    // <React.StrictMode>
        <SimpleForm></SimpleForm>
    // </React.StrictMode>
)
