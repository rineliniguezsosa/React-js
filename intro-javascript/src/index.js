import { findhero } from "./bases/08-import-export";

const promesa = new Promise((resolve,reject)=>{
    setTimeout(() => {  
        const myhero = findhero(2);
        resolve(myhero)
    },2000);
})

promesa
    .then(dato =>{
        console.log(dato);
    })

const getheroebyid = () =>{
    
}