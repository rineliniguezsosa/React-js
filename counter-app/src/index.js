import React from 'react';
import ReactDOM from 'react-dom/client';
import { Firstapp } from './Firstapp';
//import { App } from './Holamundo';
import './estilos.css'


const root = ReactDOM.createRoot(document.getElementById('root')) //REACT 18 YA  NO ES POSIBLE HACER EL RENDER DIRECTAMENTE
root.render(<Firstapp></Firstapp>) //REACT 18

