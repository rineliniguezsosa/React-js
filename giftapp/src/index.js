import React from 'react';
import ReactDOM from 'react-dom/client';
import { Giftexpertapp } from './Giftexpertapp';

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>

    </React.StrictMode>
)
