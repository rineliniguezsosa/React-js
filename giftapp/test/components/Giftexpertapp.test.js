import { render,screen,fireEvent } from "@testing-library/react";
import { Giftexpertapp } from "../../src/Giftexpertapp";

describe('Pruebas en el componente <Giftexpertapp/>', () => {
  test('Debe de coincidir con el snapshot', () => {
    const { container } = render(<Giftexpertapp></Giftexpertapp>)
    expect(container).toMatchSnapshot();
  });

  test('El state inicial debe ser gon', () => {
    render(<Giftexpertapp/>)

    expect(screen.getByText('Gon')).toBeTruthy();
  }); 

  test('Cambiando el valor de la caja de texto y debe de coincidir', () => {
    render(<Giftexpertapp/>)
    const input = screen.getByRole('textbox')
    fireEvent.input(input,{target:{value:'pac man'}})
    expect(input.value).toBeTruthy();
    screen.debug()
  });
});