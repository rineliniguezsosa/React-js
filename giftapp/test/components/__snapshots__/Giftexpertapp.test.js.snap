// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <Giftexpertapp/> Debe de coincidir con el snapshot 1`] = `
<div>
  <h1>
    Giftexpertapp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar"
      type="text"
      value=""
    />
  </form>
  <button>
    Agregar una categoria
  </button>
  <h1>
    Gon
  </h1>
  <p>
    Cargando ...
  </p>
  <div
    class="card-grid"
  />
</div>
`;
